### PR TITLE
feat: messaging + http client

### PR DESCRIPTION
**Pull request by rkrishn7**
### ↩️ Depends On

Link any other PRs here.

### 🧪 QA

- List the steps to test the changes here.

### 🛑 Risk Analysis

- [x] Does it update dependencies?

### CODE REVIEW

File: src/content-script/actions/party.ts [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
+    type: PartyActions.SET_PARTY,
+    ...details,
   };
 };
 
-export const setJoinUrl = (joinUrl: string | null) => {
-  return {
-    type: PartyActions.SET_JOIN_URL,
-    joinUrl,
+export const joinParty = ({ hash, isHost }: any) => {
```
Comment (zzulanas) [inline src/content-script/actions/party.ts:18]: so is this changing the join_URL to be a hash for the server side? or client side as well?

File: src/content-script/reducers/party.ts [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
 
 const party = (state: PartyState = initialState, action: Action) => {
   switch (action.type) {
-    case PartyActions.SET_ROOM_ID:
-      return {
-        ...state,
-        roomId: action.roomId,
-      };
-    case PartyActions.SET_JOIN_URL:
+    case PartyActions.SET_PARTY:
```
Comment (zzulanas) [inline src/content-script/reducers/party.ts:21]: the rename is gonna break some of the code for page transitions, I'll try and fix that after this is approved

File: src/content-script/actions/party.ts (hunk 1 above)
Comment (rkrishn7) [reply src/content-script/actions/party.ts:18]: > so is this changing the join_URL to be a hash for the server side? or client side as well?

The hash and join_url are generated and stored on the server.

**Review by zzulanas — APPROVED** [review]
Looks good on my end, I'll have to refactor some of the page-transitions code but that should be ezpz.

**Pull request merged**